x/profile/keeper/msg_server_update_coordinator_description.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/spn/x/profile/types"
+)
+
+func (k msgServer) UpdateCoordinatorDescription(
+	goCtx context.Context,
+	msg *types.MsgUpdateCoordinatorDescription,
+) (*types.MsgUpdateCoordinatorDescriptionResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if the coordinator address is already in the store
+	coordByAddress, found := k.GetCoordinatorByAddress(ctx, msg.Address)
+	if !found {
+		return &types.MsgUpdateCoordinatorDescriptionResponse{},
+			sdkerrors.Wrap(types.ErrCoordAddressNotFound, msg.Address)
+	}
+
+	coord := k.GetCoordinator(ctx, coordByAddress.CoordinatorId)
+	if (coord == types.Coordinator{}) {
+		panic("a coordinator address is associated to a non-existent coordinator ID")
```

Comment:
if there is a chance that coordinator will not be exists in a handler, we should return with an error instead to let client know what actually did went wrong.

Reply:
The coordinator ID returned in `GetCoordinatorByAddress` must always exist in the store. If it doesn't, this means that the blockchain logic is broken

Reply:
Hm, what does it mean blockchain logic is broken, can you demonstrate me a case how this might happen?

Reply:
The state of the blockchain can only be modified through the defined messages. If we have an impossible state from the defined messages it means the blockhain logic is broken.
In the case, wit our messages `MsgCreateCoordinator`, `MsgUpdateCoordinatorAddress`, etc... a `Coordinator` in the store is ALWAYS associated with a `CoordinatorByAddress`, if one exists without the other, it means one of the message has a vulnerability